File: silver/1966/c++/main.cpp
#include <iostream>
#include <queue>

int main()
{
    int num;
    std::cin >> num;
    

    for (int n = 0; n < num; n++) {
        std::queue<std::pair<int, int>> que;
        std::priority_queue<int> pq;
        std::pair<int, int> data;
        int max, target;
        std::cin >> max >> target;

        for (int i = 0; i < max; i++) {
            std::cin >> data.first;
            data.second = i;
            que.push(data);
            pq.push(data.first);
        }

        for (int i = 0; i < max; i++) {
            while (pq.top() != que.front().first) {
                que.push(que.front());
                que.pop();
            }
            if (que.front().second == target) {
                std::cout << i +1 << '\n';
                break;
            }
            pq.pop();
            que.pop();
        }
    }
    return 0;
}

File: silver/1966/python/main.py

if (__name__ == "__main__"):
    n = int(input())

    for _ in range(n):
        queue = list()
        pq = list()

        max, target = map(int, input().split())

        arr = list(map(int, input().split()))
        for i in range(max):
            queue.append((arr[i], i))
            pq.append(arr[i])
        pq.sort(reverse=True)
        for i in range(max):
            while (pq[0] != queue[0][0]):
                queue.append(queue.pop(0))
            if (queue[0][1] == target):
                print(i +1)
                break
            queue.pop(0)
            pq.pop(0)
